### ISSUE TITLE
Unpack `**kwargs` that hiding actual parameters for the Query Compiler methods

### ISSUE THREAD

**Issue by dchigarev** [modin-project/modin#3108]
Many of the `BaseQueryCompiler` methods are hiding actual arguments by using `*args` and `**kwargs` instead of them. They should be spread into actual parameters.

Example of function hiding parameters:
https://github.com/modin-project/modin/blob/a3ddf2f01163a312416d2a8bc456ba9582ae9b4d/modin/backends/base/query_compiler.py#L544-L550
`BaseQueryCompiler.count` method takes `**kwargs` instead of actual parameters that are passed from the front-end:
https://github.com/modin-project/modin/blob/a3ddf2f01163a312416d2a8bc456ba9582ae9b4d/modin/pandas/base.py#L970-L972

**Comment by RehanSD** [comment]
@dchigarev has this issue been resolved? I checked the `BaseQueryCompiler` and it seems like this is still happening, so should we assign someone to this?

**Comment by mvashishtha** [comment]
@RehanSD we still do this in many query compiler functions at Modin version 88f7b27d32b099c508e93251cafe503f2908d426.

**Comment by dchigarev** [comment]
@RehanSD yes, it's still relevant. As @mvashishtha noticed there are still numerous of qc methods that still follow this `**kwargs` pattern. However, the issue is not critical at all and thus should be low prioritized.

> so should we assign someone to this?

This could be a "good first issue" for a new-comers, however, as I believe we have more prioritized tasks now, so I would prefer to focus on them, but not on the very low-prioritized ones.
